price_pusher/src/injective.ts

```diff
@@ -0,0 +1,111 @@
+import { HexString } from "@pythnetwork/pyth-common-js";
+import { ChainPricePusher, PriceInfo, PriceListener } from "./interface";
+import { DurationInSeconds } from "./utils";
+import { PriceConfig } from "./price-config";
+import { ChainGrpcWasmApi } from "@injectivelabs/sdk-ts";
+
+type PriceQueryResponse = {
+  price_feed: {
+    id: string;
+    price: {
+      price: string;
+      conf: string;
+      expo: number;
+      publish_time: number;
+    };
+  };
+};
+
+// this use price without leading 0x
+export class InjectivePriceListener implements PriceListener {
```

Comment:
I see that this is very similar to EVM. Maybe it's good to extract common identical methods to `PriceListener`.

Reply:
agree with this. i think the latestPriceInfo / update / get methods could be moved up to priceListener

I wouldn't move the pollPrices method even though it is very similar to the evm method, as you want to give implementations the flexibility to implement the reading logic in different ways.